rfcs/0123-flake-names.md

```diff
@@ -0,0 +1,54 @@
+---
+feature: flake-names
+start-date: 2022-03-12
+author: Anselm Schüler
+co-authors: None
+shepherd-team: None
+shepherd-leader: None
+related-issues: None
+---
+
+# Summary
+[summary]: #summary
+
+Flakes can declare the field `name`.  
+It represents the name of the flake.  
+The derivations for a flake are no longer called `source`, but use the flake name.
+
+# Motivation
+[motivation]: #motivation
+
+Flake-centric workflows often end up with a lot of derivations named “source”, and it’s difficult to navigate this.
+Also, the discoverability and usability of flakes needs to be improved. Current commands mostly show technical information. This would be a step in the right direction.
+
+# Detailed design
+[design]: #detailed-design
+
+A new supported property for flakes is introduced, `name`.  
+Running `nix flake metadata` on a flake that declares this field displays it at the top.  
+The derivation that contains the flake’s content is called `flake-source-${name}` or, if a short revision identifier is available, `flake-source-${name}-${shortRev}`.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+A flake that is in the registry should have its name match the registry identifier.
```

Comment:
Wouldn't that prevent the use of alternate names, or having multiple versions of the same flake in the registry?
(I use both)

Reply:
That’s a good point

Reply:
Maybe it’d be a good rule for the global registry, though, but it’d probably be followed anyways, and would only hinder edge-cases